### PR TITLE
Add dashboard list panel type

### PR DESCRIPTION
**Pull request by stevenluan**

### CODE REVIEW

File: grafana/panels/dashboard_list.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    var self = this;
+
+    this.state = {
+        title: 'dashboard list',
+        error: false,
+        span: 3,
+        editable: true,
+        type: 'dashlist',
+        isNew: true,
+        id: 7,
```
Comment (eculver) [inline grafana/panels/dashboard_list.js]: This should not be part of the state. The backend should fill this in.

File: grafana/panels/dashboard_list.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    this.state.title = opts.title || this.state.title;
+    this.state.id = opts.id || generateGraphId();
+    this.state.span = opts.span || 3;
+    
+    // finally add to row/dashboard if given
+    if (opts.row && opts.dashboard) {
+        opts.row.addPanel(this);
+        opts.dashboard.addRow(opts.row);
+    }
+    for (var k in opts) {
```
Comment (eculver) [inline grafana/panels/dashboard_list.js]: I think this should come after the state defaults. See `./grafana/panels/graph.js` as a good example.

Something like this:
```
    // Overwrite defaults with custom values
    Object.keys(opts).forEach(function eachOpt(opt) {
        self.state[opt] = opts[opt];
    });
```

File: grafana/panels/dashboard_list.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        mode: 'search',
+        query: 'dashboard list',
+        limit: 10,
+        tags: [],
+        links: []
+    };
+
+    this.state.title = opts.title || this.state.title;
+    this.state.id = opts.id || generateGraphId();
+    this.state.span = opts.span || 3;
```
Comment (eculver) [inline grafana/panels/dashboard_list.js]: Also, these pretty much can go away if opts are merged over the defaults.

**Comment by albertyw** [comment]
Merged in #50

**Pull request closed without merging**